### PR TITLE
Hotfix/workers base refactor

### PR DESCRIPTION
**Pull request by siepet**
Using Szymon tips to make a parent Base base class, I did it. It is inherited in other base classes per integration.

### CODE REVIEW

File: app/jobs/rollbar/base.rb [hunk 1]
```diff
@@ -1,29 +1,21 @@
...
-    def user_repo
-      @user_repo ||= UserRepository.new(data_guru.members.all)
+    def fetch_basic_teams
+      @api_teams ||= RollbarIntegration::Team.all_from_api(api, user_repo)
     end
 
-    def dataguru_teams
-      @dataguru_teams ||= RollbarIntegration::Team.all_from_dataguru(data_guru.rollbar_teams)
+    def api_teams
+      @api_teams = RollbarIntegration::Team.add_projects(fetch_basic_teams, api)
```
Comment (Szeliga) [inline app/jobs/rollbar/base.rb]: I think you forgot `||=`

Comment (siepet) [inline app/jobs/rollbar/base.rb]: If i did `||=` then it wouldn't return teams with projects. I use same variable to first create teams with people, and then build teams with projects.

Comment (Szeliga) [inline app/jobs/rollbar/base.rb]: true that

**Comment by Szeliga** [comment]
Good to merge :+1:

**Comment by lubieniebieski** [comment]
:blue_heart: go for it:)

**Pull request merged**